src/JHipsterNetSampleApplication/Data/Extensions/DbSetExtensions.cs

```diff
@@ -33,5 +34,63 @@ private static PropertyInfo GetKeyProperty(Type type)
             //TODO complete with FluentAPi
             return null;
         }
-    }
+
+        public static void RemoveNavigationProperty<TEntity, TOwnerEntity>(this DbSet<TEntity> receiver, TOwnerEntity ownerEntity, object id)
+            where TEntity : class
+            where TOwnerEntity : class
+        {
+            bool success;
+            var receiverObjects = receiver.ApplyWhere(ownerEntity.GetType().Name + "Id", id, out success);
+
+            if (success) {
+                foreach (TEntity receiverObject in receiverObjects) {
+                    receiver.Remove(receiverObject);
+                }
+            }            
+        }
+
+        private static IQueryable<T> ApplyWhere<T>(this IQueryable<T> source, string propertyName, object propertyValue, out bool success)
+            where T : class
+        {
+            // 1. Retrieve member access expression
+            success = false;
```

Comment:
Throw a technical Exception you have to catch on `RemoveNavigationProperty`  or play with a `java.util.Optional` .Net equivalent rather than using `out`